core/services/job/job_orm_test.go

```diff
@@ -849,6 +851,57 @@ func Test_FindPipelineRunIDsByJobID(t *testing.T) {
 
 		assert.Equal(t, run.ID, runIDs[0])
 	})
+
+	jobs := make([]job.Job, 10)
+	for j := 0; j < len(jobs); j++ {
+		_, bridge := cltest.MustCreateBridge(t, db, cltest.BridgeOpts{}, config)
+		_, bridge2 := cltest.MustCreateBridge(t, db, cltest.BridgeOpts{}, config)
+		jobID := uuid.NewV4().String()
+		key, err := ethkey.NewV2()
+
+		require.NoError(t, err)
+		jb, err = ocr.ValidatedOracleSpecToml(cc,
+			testspecs.GenerateOCRSpec(testspecs.OCRSpecParams{
+				JobID:              jobID,
+				Name:               fmt.Sprintf("Job #%v", jobID),
+				DS1BridgeName:      bridge.Name.String(),
+				DS2BridgeName:      bridge2.Name.String(),
+				TransmitterAddress: address.Hex(),
+				ContractAddress:    key.Address.Address().String(),
+			}).Toml())
+
+		require.NoError(t, err)
+
+		err = orm.CreateJob(&jb)
+		require.NoError(t, err)
+		jobs[j] = jb
+	}
+
+	for i, j := 0, 0; i < 2500; i++ {
+		mustInsertPipelineRun(t, pipelineORM, jobs[j])
+		j++
+		if j == len(jobs) {
+			j = 0
+		}
+	}
+
+	// Internally these queries are batched by 1000, this tests case requiring concatenation
+	//  of more than 1 batch
+	t.Run("with batch concatenation limit 10", func(t *testing.T) {
```

Comment:
Would it make more sense to pass in the batch size as an argument, so it isn't necessary to insert so many records in the test?

Reply:
Yeah, I don't like changing the interface just for a test, but I think in this case it's probably worth it.